.github/workflows/main.yml

```diff
@@ -0,0 +1,32 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.7
+        bundler-cache: true
+    - name: Build
+      run: bundle exec jekyll build
+    - name: Deploy
+      # if: github.event_name == 'push'
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: docs
+        user_name: 'github-actions[bot]'
+        user_email: 'github-actions[bot]@users.noreply.github.com'
```

Comment:
Does this end up making a second commit based on this commit?

Reply:
Yeah commits in `gh-pages` branch are separate, since they're for autogenerated files not for authored ones.

You can see an example deployment commit here: https://github.com/WebAssembly/website/commit/fef1b8cf252dc1aa9bef396172bc5959b7a991da